File: adventuraclick_mobile/lib/model/travel.dart
import 'package:adventuraclick_mobile/data/mockResponse.dart';
import 'package:adventuraclick_mobile/model/included_item.dart';
import 'package:adventuraclick_mobile/model/location.dart';
import 'package:adventuraclick_mobile/model/travel_information.dart';
import 'package:adventuraclick_mobile/model/travel_type.dart';
import 'package:json_annotation/json_annotation.dart';
part 'travel.g.dart';

@JsonSerializable()
class Travel {
  int? travelId;
  String? name;
  String? status;
  String? image;
  String? description;
  double? price;
  int? numberOfNights;
  DateTime? date;
  int? locationId;
  int? travelTypeId;
  TravelType? travelType;
  Location? location;
  List<IncludedItem>? includedItems;
  List<TravelInformation>? travelInformations;

  Travel();

  factory Travel.fromJson(Map<String, dynamic> json) => _$TravelFromJson(json);

  /// Connect the generated [_$TravelToJson] function to the `toJson` method.
  Map<String, dynamic> toJson() => _$TravelToJson(this);
}

File: AdventuraClick.Model/Travel.cs
namespace AdventuraClick.Model
{
    public partial class Travel
    {
        public int TravelId { get; set; }
        public string Name { get; set; } = null!;
        public byte[]? Image { get; set; }
        public int NumberOfNights { get; set; }
        public string? Description { get; set; }
        public float Price { get; set; }
        public string? Status { get; set; }
        public int? TravelTypeId { get; set; }
        public int? LocationId { get; set; }
        public int? TravelInformationId { get; set; }
        public virtual ICollection<TravelInformation> TravelInformations { get; set; }
        public virtual TravelType? TravelType { get; set; }
        public virtual Location? Location { get; set; }
        public List<IncludedItem> IncludedItems { get; set; }

    }
}